task-2-SERVLETS/Self-check.md

```diff
@@ -0,0 +1,13 @@
+1. What is the Servlet Container? List some of them.
+2. What is the life-cycle of a servlet?
+3. What are life-cycle methods for a servlet do you know and what is their functions?
+4. What is REST?
+5. What HTTP methods are used according to the REST?
+6. Describe what http-statuses are used in the responses and what server behaviour they mean according to the REST.
+7. What is the difference between PUT and PATCH methods?
+8. What is the Servlet Listener? What is the purpose of it?
+9. What is the Servlet Filter? What is the purpose of it?
+10. Describe the behaviour when a request is sent to the server?
```

Comment:
Suppose this would be better:
10. Describe a server behavior when a request is sent to the server?

Reply:
You can tap on 'insert a suggestion' or ctrl-g
```suggestion
10. Describe a server behaviour when a request is sent to the server?
```

Reply:
And then even commit it)